lib/components/Modal/base/index.tsx

```diff
@@ -32,172 +33,130 @@ export interface ModalProps extends HTMLAttributes<HTMLDivElement> {
-                          <IconButton
-                              variant="tertiary"
-                              color="black-white"
-                              size="lg"
-                              className="quill-modal__close-icon"
-                              onClick={toggleHandler}
-                              icon={<LabelPairedXmarkMdBoldIcon />}
-                          />
-                      )}
-
-                <div className="quill-modal__content-wrapper">{children}</div>
-                {hasFooter && (
-                    <div className="quill-modal__button-wrapper">
-                        {showPrimaryButton && (
-                            <Button
-                                color={buttonColor}
-                                fullWidth
-                                size="lg"
-                                label={primaryButtonLabel}
-                                disabled={isPrimaryButtonDisabled}
-                                onClick={primaryButtonFunctionHandler}
-                            />
-                        )}
-                        {showSecondaryButton && (
-                            <Button
-                                color="black-white"
-                                fullWidth
-                                size="lg"
-                                label={secondaryButtonLabel}
-                                disabled={isSecondaryButtonDisabled}
-                                variant="secondary"
-                                className="quill-modal__button"
-                                onClick={secondaryButtonFunctionHandler}
-                            />
-                        )}
-                    </div>
-                )}
-            </div>
-        </div>,
-        modalRoot,
+                <div
+                    className={clsx("quill-modal__container", className)}
+                    onClick={(e) => e.stopPropagation()}
+                >
+                    {showCrossIcon && (
+                        <IconButton
+                            variant="tertiary"
+                            color="black-white"
+                            size="lg"
+                            className="quill-modal__close-icon"
+                            onClick={handleClose}
+                            icon={<LabelPairedXmarkMdBoldIcon />}
+                        />
+                    )}
+                    <ModalContent />
+                </div>
+            </div>,
+            modalRoot,
+        )
+    ) : (
+        <ActionSheet.Root isOpen={isOpened} onClose={handleClose}>
+            <ActionSheet.Portal
+                {...rest}
+                showHandlebar={showHandleBar}
+                handleBarPosition="absolute"
+                disableCloseOnOverlay={disableCloseOnOverlay}
+                portalId={portalId}
+            >
+                <ModalContent />
+            </ActionSheet.Portal>
+        </ActionSheet.Root>
```

Comment:
We can't make Modal component to fully action sheet because there is some different design for modal component like padding, h4 title (h5 in actionsheet), image header and etc. But the portal and handle swipe close can be use actionsheet portal, in this case we can remove duplicated swipe function in modal component.